reactor-core/src/main/java/reactor/core/publisher/SinksSpecs.java

```diff
@@ -34,18 +35,16 @@
 	final Many<T>       sink;
 	final ContextHolder contextHolder;
 
-	volatile     Throwable                                                  error;
+	volatile int wip;
 	@SuppressWarnings("rawtypes")
-	static final AtomicReferenceFieldUpdater<SerializedManySink, Throwable> ERROR =
-			AtomicReferenceFieldUpdater.newUpdater(SerializedManySink.class, Throwable.class, "error");
+	static final AtomicIntegerFieldUpdater<SerializedManySink> WIP =
+			AtomicIntegerFieldUpdater.newUpdater(SerializedManySink.class, "wip");
 
 	volatile Thread lockedAt;
 	@SuppressWarnings("rawtypes")
 	static final AtomicReferenceFieldUpdater<SerializedManySink, Thread> LOCKED_AT =
 			AtomicReferenceFieldUpdater.newUpdater(SerializedManySink.class, Thread.class, "lockedAt");
 
-	volatile boolean done;
```

Comment:
I know I said that one wasn't probably useful offband, but while reviewing I thought about one benefit: it allows a more meaningful error code to be returned before checking for unsynchronized access: we _can_ relax on the serialized access condition if we know that the sink is terminated anyway (since fail fast to `FAIL_TERMINATED` wouldn't result in any spec violation downstream).

Reply:
We _could_, but I had a feeling that it overcomplicates the implementation and preferred to stick to as trivial implementation as possible. Also, failing fast would hide the fact of a non-serialized access that is very racy and should ideally be detected ASAP with an increased chance of being triggered from tests. WDYT?